extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/JDBCExtractionNamespaceCacheFactory.java

```diff
@@ -42,32 +44,34 @@
 /**
  *
  */
-public class JDBCExtractionNamespaceCacheFactory
+public final class JDBCExtractionNamespaceCacheFactory
     implements ExtractionNamespaceCacheFactory<JDBCExtractionNamespace>
 {
   private static final Logger LOG = new Logger(JDBCExtractionNamespaceCacheFactory.class);
-  private final ConcurrentMap<String, DBI> dbiCache = new ConcurrentHashMap<>();
+  private final ConcurrentMap<CacheScheduler.EntryImpl<JDBCExtractionNamespace>, DBI> dbiCache =
+      new ConcurrentHashMap<>();
 
   @Override
-  public String populateCache(
-      final String id,
+  @Nullable
+  public CacheScheduler.VersionedCache populateCache(
       final JDBCExtractionNamespace namespace,
+      final CacheScheduler.EntryImpl<JDBCExtractionNamespace> entryId,
       final String lastVersion,
-      final Map<String, String> cache
-  ) throws Exception
+      final CacheScheduler scheduler
+  )
   {
     final long lastCheck = lastVersion == null ? JodaUtils.MIN_INSTANT : Long.parseLong(lastVersion);
-    final Long lastDBUpdate = lastUpdates(id, namespace);
+    final Long lastDBUpdate = lastUpdates(entryId, namespace);
     if (lastDBUpdate != null && lastDBUpdate <= lastCheck) {
-      return lastVersion;
+      return null;
     }
     final long dbQueryStart = System.currentTimeMillis();
-    final DBI dbi = ensureDBI(id, namespace);
+    final DBI dbi = ensureDBI(entryId, namespace);
     final String table = namespace.getTable();
     final String valueColumn = namespace.getValueColumn();
     final String keyColumn = namespace.getKeyColumn();
 
-    LOG.debug("Updating [%s]", id);
+    LOG.debug("Updating %s", entryId);
```

Comment:
CacheScheduler.Entry and EntryImpl's `toString()` returns strings of the form `"namespace: [...]"`, it tries to mimic some old logging strings. So this line should be like `Updating namespace: [...]`